chingu-voyages/v51-tier3-team-33
# Gary/feature/group invite link

Please only view this request after the request gary/feature/add-receipt as this builds on top of it.

## Description
This commit is for the inclusion of group links. I changed the Group model where each group now has a unique 7 digit link when created. This is passed over to the frontend where it can later be used for a user to join a group upon clicking a link. I also changed the api route when a user to be added to the group to use the invite link of a group instead of using the group id.

## Related to the following Jira/Trello issue 
Im not sure if Radhika made one, but if i find it later i will link it here.

## Main changes explained:
- Modified Group model to include addition of invite_link field.
- Modified Group/route.ts to include generation of unique invite links and include generated link in creation of new group.
- Moved logic of adding user to group from groups/add_user/route ts to invite/[inviteLink] to handle adding to the user to group associated with invite link.

## How to test: 
0. Easiest way to test this change is to use Postman or something similar.
1. check into current branch.
2. do `pnpm install` and `pnpm run dev` to run this PR locally.
3. Create a post request on Postman to add a new group to the database, and verify the inclusion of invite_link upon creation.
4.  Create another request to add an existing user to the new group. The route should be formatted as localhost:3000/invite/*insert invite id of new group here*. The id of the user should also be passed in the body of the request as an object. Ex: { "user_id": "66f83e89dcdaaa757d8642b9"} 
5. Verify that the user was successfully added to the group by seeing the user information under the members section of the response from the request.
